postgresmodels/posture.go

```diff
@@ -0,0 +1,136 @@
+package postgresmodels
+
+import (
+	"time"
+
+	"github.com/lib/pq"
+	"gorm.io/datatypes"
+)
+
+// TODO: add explicit column names, add validation
+
+type Cluster struct {
+	BaseModel
+	GUID         string `gorm:"primaryKey"`
+	CustomerGuid string
+	//Location      string
+	CloudProvider string
+	FullName      string
+	ShortName     string
+	PrefixName    string
+}
+
+type ClusterPostureReport struct {
+	ReportGUID        string `gorm:"primaryKey"`
+	ClusterGUID       string
+	Cluster           Cluster `gorm:"foreignKey:ClusterGUID"`
+	Score             float32
+	Timestamp         time.Time
+	WorkerNodeCount   int
+	KubescapeVersion  string
+	KubernetesVersion string
+	HelmChartVersion  string
+	//RegoLibraryVersion  string
+	ControlsSummary     ControlsSummary      `gorm:"foreignKey:ReportGUID"`
+	ControlsScanResults []ControlScanResult  `gorm:"foreignKey:ReportGUID"`
```

Comment:
we shouldn't have such constraints so we can be agnostic to the insertion order.